### PR TITLE
Use email.utils to parse the from domain(s)

### PR DESCRIPTION
**Pull request by adam-iris**
This probably isn't ready to merge!

I added a unit test, and it seems outside the scope of this fix but the test message returns `dkim=fail`. I probably did something wrong in how I generated the signature (although the body b= key looks right) but I could be missing something deeper.

### CODE REVIEW

**Comment by niftylettuce** [comment]
Bump

**Comment by niftylettuce** [comment]
Would gladly award a bounty if this could be finished

**Comment by adam-iris** [comment]
Sorry I got pulled off this project. 

I think the code is ready, the only hiccup was that I couldn’t add a working unit test because I couldn’t generate a proper key to test against. That’s really outside the scope of the fix, though, so it doesn’t seem critical.

**Comment by kitterma** [comment]
Please do add tests.  If you look in the authheaders/test directory there is a sample key you can use there.  Have a look at the other tests to see how it's used.

**Comment by adam-iris** [comment]
You can see the (failing) unit test in the history, the issue was that I had to create a new message for the test, but I wasn't able to generate a DKIM signature for it that validated. That validation is really outside the scope of the change -- which was mainly about switching to a standard package for parsing incoming email addresses -- so I wound up removing it.

I hadn't thought about it this way before, but I'd say this doesn't really require a separate unit test, because it already gets tested by the existing ones. That is, all the existing unit tests use the new code for parsing email addresses, so they adequately test whether this code is calling the library correctly. So all my test was adding was whether the library itself properly parses some edge cases, which isn't really relevant here.

TL;DR: I think this is ready to go

**Comment by niftylettuce** [comment]
Would love to see this added!

**Comment by shaunwarman** [comment]
+1

**Comment by niftylettuce** [comment]
I tested this and it's broken by the way, see my comments here #5 and #13, even with the latest recommended fixes (ref: https://github.com/forwardemail/authentication-headers).

**Comment by niftylettuce** [comment]
@adam-iris I have numerous other fixes in my branch, which you might want to sync up with yours and add to your pull request to make it easy for the ValiMail repo owners to merge.

e.g. https://github.com/ValiMail/authentication-headers/issues/13#issuecomment-670227986

See the commit history here:

https://github.com/forwardemail/authentication-headers/commits/master

**Comment by niftylettuce** [comment]
I have published another fix for yet another issue, https://github.com/forwardemail/authentication-headers/commit/1046cd72103e432f4aa2d152b75955db1d481486

**Comment by niftylettuce** [comment]
Another fix has been published, see https://github.com/ValiMail/authentication-headers/issues/18 and https://github.com/forwardemail/authentication-headers/commit/b123501bd8e5998a52b52978dc39e0f263311607.

**Comment by niftylettuce** [comment]
Another fix can be found here: <https://github.com/forwardemail/authentication-headers/commit/54d1bc2d312baa1c8ea9159d7801417749ab746a>
